Adapt default charge change score to allow for charge changing transformations

hannahbaumann opened the pull request:
Only adapt this in gufe bindings to not affect main Lomap behaviour.

- [x] Check if docs have to be updated
 ---> The scorers are not part of the API documentation?!

hannahbaumann committed:
Adapt default charge change score to allow for charge changing transformations

atravitz reviewed (changes requested):
This should have a corresponding test and news entry, otherwise looks good!

IAlibay reviewed (changes requested):
Hold on my end - I want to double check the other ways to use Lomap networks aren't being affected.

hannahbaumann committed:
Update test

hannahbaumann committed:
Add news entry

hannahbaumann commented:
@atravitz I changed a test that was checking the connection of networks when using a charge score of 0.1 to test that with the new default (charge score of 0.1) we are getting connected networks. Or should I also explicitly test for the value of the score?
I also added a news item.

atravitz commented:
The check you have is great, but I would also explicitly check that the correct value is being assigned, since we'd want tests to break if the actual value is getting mutated somewhere and turning out to be not 0.1.

Other than that, this looks good to me!

IAlibay commented on news/charge_change_score.rst:
Rever isn't setup yet, so let's merge after #86

IAlibay reviewed (approved):
Sorry about the delay, took me a while to realise all of scorers.py is a duplicate of what's in the MCS file (we should fix that later).

atravitz commented on news/charge_change_score.rst:
```diff
@@ -0,0 +1,24 @@
+
+**Changed:**
+
+* Changed the default `charge_changes_score` from 0.0 to 0.1 in the gufe bindings to enable connected networks for 
```
```suggestion
* Changed the default ``charge_changes_score`` from 0.0 to 0.1 in the gufe bindings to enable connected networks for 
```

rst needs two back ticks for code formatting (otherwise it's italics)

hannahbaumann committed:
Update news/charge_change_score.rst

Co-authored-by: Alyssa Travitz <31974495+atravitz@users.noreply.github.com>

hannahbaumann committed:
Test default charge change score

hannahbaumann committed:
Add default_lomap_score to api docs

hannahbaumann committed:
Small fix

atravitz reviewed (approved):
thanks @hannahbaumann - the docs are building correctly for me locally, so I consider this a readthedocs headache that I will address in a follow-up (likely [here](https://github.com/OpenFreeEnergy/Lomap/pull/90)).

atravitz committed:
Merge branch 'main' of github.com:OpenFreeEnergy/Lomap into change_charge_score_default

atravitz commented:
docs are fixed!
